e2e-tests/spring-rest-openapi-v2/src/main/java/com/foo/rest/examples/spring/security/accesscontrol/deleteput/SimpleAccessControlRest.java

```diff
@@ -0,0 +1,150 @@
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        // anonymous users cannot create resources
+        if (authentication.getPrincipal().equals("anonymousUser")) {
+
+            return new ResponseEntity<>("Only authenticated users can create resources ", HttpStatus.UNAUTHORIZED);
+        }
+        // the user is not anonymous
+        else {
+
+            String creatorUser = authentication.getName();
+
+            ResourceDto newResource = new ResourceDto();
+            newResource.stringValue = stringValue;
+            newResource.integerValue = integerValue;
+            newResource.booleanValue = booleanValue;
+
+            // if the resource exists and the user sending this request is the resource owner, overwrite the resource
+            if (resources.containsKey(x)) {
+
+                if (resourceOwners.get(x).equals(creatorUser)) {
+
+                    resources.put(x, newResource);
+                    return new ResponseEntity<>("Resource has been successfully recreated by its owner ",
+                            HttpStatus.OK);
+                }
+                else {
+                    return new ResponseEntity<>("Only the owner can recreate the resource ", HttpStatus.UNAUTHORIZED);
+                }
+            }
+            else {
+
+                // if the user has role CREATOR, create the resource
+                if (authentication.getAuthorities().contains(new SimpleGrantedAuthority("ROLE_CREATOR"))) {
+                    resources.put(x, newResource);
+                    resourceOwners.put(x, creatorUser);
+
+                    return new ResponseEntity<>("New resource has been successfully created ",
+                            HttpStatus.OK);
+                }
+                else {
+                    return new ResponseEntity<>("Only users with creator role can create resources ",
+                            HttpStatus.UNAUTHORIZED);
+                }
+
+            }
+        }
+    }
+
+    // put is to update an existing resource, this endpoint is vulnerable in the sense that
+    // it does not check the owner of the resource before updating it
+    @PutMapping(value = "/{x}")
+    public ResponseEntity modifyResource(@PathVariable("x") String x, @RequestParam String stringValue, @RequestParam int integerValue, @RequestParam boolean booleanValue ) {
+
+        ResourceDto newResource = new ResourceDto();
+        newResource.stringValue = stringValue;
+        newResource.integerValue = integerValue;
+        newResource.booleanValue = booleanValue;
+
+        resources.put(x, newResource);
+
+        return new ResponseEntity<>("Modified the resource successfully ",
+                HttpStatus.OK);
+
+    }
+
+    // delete endpoint checks the owner of the resource to make sure the resource cannot be deleted
+    @DeleteMapping(value = "/{x}")
+    public ResponseEntity deleteResource(@PathVariable("x") String x) {
```

Comment:
Authentication object can be autowired here as input. see examples

Reply:
Hi Andrea,

Thanks for your comment, not sure if I understood this one well.